litestar/security/jwt/token.py

```diff
@@ -85,8 +90,35 @@ def decode(cls, encoded_token: str, secret: str | dict[str, str], algorithm: str
         Raises:
             NotAuthorizedException: If the token is invalid.
         """
+
+        def base64url_decode(code: str) -> bytes:
+            padding = "=" * (4 - (len(code) % 4))
+            return base64.urlsafe_b64decode(code + padding)
+
+        if isinstance(secret, str):
+            converted_secret: AllowedPublicKeys | str | bytes = secret
+        else:
+            if secret["kty"] == "RSA":
+                n = int.from_bytes(base64url_decode(secret["n"]), byteorder="big")
+                e = int.from_bytes(base64url_decode(secret["e"]), byteorder="big")
+                converted_secret = rsa.RSAPublicNumbers(e, n).public_key(default_backend())
+            elif secret["kty"] == "EC":
+                x = int.from_bytes(base64url_decode(secret["x"]), byteorder="big")
+                y = int.from_bytes(base64url_decode(secret["y"]), byteorder="big")
+                converted_secret = ec.EllipticCurvePublicNumbers(x, y, ec.SECP256R1()).public_key(default_backend())
+            elif secret["kty"] == "OKP" and secret["crv"] == "Ed25519":
+                x = base64url_decode(secret["x"])
+                converted_secret = ed25519.Ed25519PublicKey.from_public_bytes(x)
+            elif secret["kty"] == "OKP" and secret["crv"] == "Ed448":
+                x = base64url_decode(secret["x"])
+                converted_secret = ed448.Ed448PublicKey.from_public_bytes(x)
+            else:
+                raise TypeError("The secret is not a form of allowed public key.")
```

Comment:
These changes seem to be unrelated to the switch to pyjwt. If this is a feature you want to add, this should be done in a separate PR.

Reply:
@provinzkraut Some details from the author would be good, but I do believe this is park of the JWK/backend implementations here: https://github.com/mpdavis/python-jose/tree/master/jose